### PR TITLE
ethereum client unit tests

### PR DESCRIPTION
**Pull request by Holymir**
Signed-off-by: Ventsislav Tsochev <ventsi@limechain.tech>

**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: app/clients/ethereum/client_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	assert.NotNil(t, nodeClient)
+}
+
+func TestGetClient(t *testing.T) {
+	nodeClient := NewClient(tc.TestConfig.Validator.Clients.Ethereum)
+	assert.NotNil(t, nodeClient)
+
+	client := nodeClient.GetClient()
+	assert.NotNil(t, client)
+}
```
Comment (georgiyazovaliiski) [inline app/clients/ethereum/client_test.go:41]: You can initialize a "mocked" client as a global variable and use it as expectation when asserting the client returned from `NewClient` and `GetClient`

File: app/clients/ethereum/client_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	chainID := nodeClient.ChainID()
+	expectedChainID := big.NewInt(3)
+	assert.Equal(t, expectedChainID, chainID)
+}
+
+func TestValidateContractDeployedAt(t *testing.T) {
+	nodeClient := NewClient(tc.TestConfig.Validator.Clients.Ethereum)
+	contractAddress := "0xbaA7610B498f7527D58bDced6046a8e7202180FD"
+	address, err := nodeClient.ValidateContractDeployedAt(contractAddress)
+	assert.NotNil(t, address)
```
Comment (georgiyazovaliiski) [inline app/clients/ethereum/client_test.go:101]: Is there any way to assert the address value?

File: app/clients/ethereum/client_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	nodeClient := NewClient(tc.TestConfig.Validator.Clients.Ethereum)
+	transactionHash := "0x9634c0496039bee73555688efd5943a5cf86caa0d4826edb0a7d7642fc8e0392"
+	nodeClient.WaitForTransaction(transactionHash, onSuccess, onRevert, onError)
+}
+
+func TestWaitForTransactionError(t *testing.T) {
+	nodeClient := NewClient(tc.TestConfig.Validator.Clients.Ethereum)
+	transactionHash := "0x9634c0496039bee73555688efd5943a5cf86caa0d4826edb0a7d7642fc8e0390"
+	nodeClient.WaitForTransaction(transactionHash, onSuccess, onRevert, onError)
+}
```
Comment (georgiyazovaliiski) [inline app/clients/ethereum/client_test.go:129]: Can we actually assert which method gets called?

**Review by georgiyazovaliiski — CHANGES_REQUESTED** [review]

File: app/clients/hedera/client_test.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+func TestNewNodeClient(t *testing.T) {
+	nodeClient := NewNodeClient(tc.TestConfig.Validator.Clients.Hedera)
+	assert.NotNil(t, nodeClient)
+}
+
+func TestGetClientForTestnet(t *testing.T) {
+	nodeClient := NewNodeClient(tc.TestConfig.Validator.Clients.Hedera)
+	client := nodeClient.GetClient()
+	assert.NotNil(t, client)
```
Comment (georgiyazovaliiski) [inline app/clients/hedera/client_test.go:36]: Here you assert that the client is not nil. Are there any properties inside the client that differ it from the other 2 tests related to getting different types of clients?

**Review by georgiyazovaliiski — APPROVED** [review]

**Comment by Daniel-K-Ivanov** [comment]
Closing this PR since the unit tests for the clients have been implemented in #303

**Pull request closed without merging**